system-int-test/src/cmr/system_int_test/data2/stac.clj

```diff
@@ -6,26 +6,44 @@
 
 (defn- href
   "Returns the link href"
-  [query-string page-num]
-  (format "%sgranules.stac?%s&page_num=%s" (url/search-root) query-string page-num))
+  ([]
+   (format "%sgranules.stac" (url/search-root)))
+  ([query-string page-num]
+   (format "%sgranules.stac?%s&page_num=%s" (url/search-root) query-string page-num)))
 
 (defn- result-map->expected-links
   "Returns the stac links for the given result map"
-  [result-map]
-  (let [{:keys [query-string page-num prev-num next-num]} result-map]
-    (remove nil?
-            [{:rel "self"
-              :href (href query-string page-num)}
-             {:rel "root"
-              :href (url/search-root)}
-             (when prev-num
-               {:rel "prev"
-                :method "GET"
-                :href (href query-string prev-num)})
-             (when next-num
-               {:rel "next"
-                :method "GET"
-                :href (href query-string next-num)})])))
+  ([result-map]
+   (result-map->expected-links result-map "GET"))
+  ([result-map method]
+   (let [{:keys [query-string page-num prev-num next-num body]} result-map]
+     (remove nil?
+             [{:rel "self"
+               :href (href query-string page-num)}
+              {:rel "root"
+               :href (url/search-root)}
+              (if (= method "GET")
+                (when prev-num
+                  {:rel "prev"
+                   :method "GET"
+                   :href (href query-string prev-num)})
+                (when prev-num
+                  {:rel "prev"
+                   :method "POST"
+                   :body body
+                   :merge true
+                   :href (href)}))
```

Comment:
Where is the `page_num` in POST? I don't think this works.

Reply:
Also what if there is combination of query string in the POT body and parameters?